test/ts/integration/examples/erc721_collateralized_simple_interest_terms_contract/scenarios/index.ts

```diff
@@ -0,0 +1,241 @@
+import { BigNumber } from "bignumber.js";
+import * as Units from "../../../../test_utils/units";
+
+import {
+    SimpleInterestContractTerms,
+    SimpleInterestParameters,
+} from "../../../../factories/terms_contract_parameters";
+import { DummyTokenContract } from "../../../../../../types/generated/dummy_token";
+import { SignedDebtOrder } from "../../../../../../types/kernel/debt_order";
+
+export const DEFAULT_REGISTER_TERM_START_ARGS = {
+    // Simple terms parameters.
+    principalAmount: Units.ether(1),
+    interestRateFixedPoint: Units.interestRateFixedPoint(2.5),
+    amortizationUnitType: new BigNumber(1),
+    termLengthUnits: new BigNumber(4),
+    // Parameters for collateralization.
+    collateralToken: "REP",
+    collateralId: new BigNumber(0),
+    // Misc parameters.
+    debtorFee: Units.ether(0.001),
+    invokedByDebtKernel: true,
+    permissionToCollateralize: true,
+    principalTokenInRegistry: true,
+    collateralTokenInRegistry: true,
+    succeeds: true,
+    isCryptoKitty: false,
+    reverts: false,
+    termsContractParameters: (terms: SimpleInterestContractTerms) =>
+        SimpleInterestParameters.pack(terms),
+};
+
+export const DEFAULT_RETURN_COLLATERAL_ARGS = {
+    principalAmount: Units.ether(1),
+    interestRateFixedPoint: Units.interestRateFixedPoint(2.5),
+    amortizationUnitType: new BigNumber(1),
+    termLengthUnits: new BigNumber(4),
+    repaymentAmount: Units.ether(1.29),
+    debtorFee: Units.ether(0.001),
+    // Parameters for collateralization.
+    collateralToken: "MET",
+    collateralId: new BigNumber(0),
+    repaymentToken: (principalToken: DummyTokenContract, otherToken: DummyTokenContract) =>
+        principalToken,
+    debtOrder: (debtOrder: SignedDebtOrder) => debtOrder,
+    principalTokenInRegistry: true,
+    collateralTokenInRegistry: true,
+    isCryptoKitty: false,
+    succeeds: true,
+    reverts: false,
+};
+
+export const DEFAULT_SEIZE_COLLATERAL_ARGS = {
+    principalAmount: Units.ether(1),
+    interestRateFixedPoint: Units.interestRateFixedPoint(2.5),
+    amortizationUnitType: new BigNumber(1),
+    termLengthUnits: new BigNumber(4),
+    repaymentAmount: Units.ether(1.29),
+    debtorFee: Units.ether(0.001),
+    // Parameters for collateralization.
+    collateralToken: "MET",
+    collateralId: new BigNumber(0),
+    repaymentToken: (principalToken: DummyTokenContract, otherToken: DummyTokenContract) =>
+        principalToken,
+    debtOrder: (debtOrder: SignedDebtOrder) => debtOrder,
+    principalTokenInRegistry: true,
+    collateralTokenInRegistry: true,
+    isCryptoKitty: false,
+    secondsSinceFill: 0,
+    succeeds: true,
+    reverts: false,
+};
+
+export const DEFAULT_REGISTER_REPAYMENT_ARGS = {
```

Comment:
you could also define a base object and unpack it, and overwrite what's different in each scenario